Capstone/Feed/HomeViewController.m

```diff
@@ -73,55 +54,72 @@ - (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText
     } else {
         isFiltered = true;
         self->filteredColleges = [[NSMutableArray alloc] init];
-        for (College *college in self->colleges) {
+        for (College *college in self->collegesAtSegment) {
             NSRange collegeName = [college.name rangeOfString:searchText options:NSCaseInsensitiveSearch];
             if (collegeName.location != NSNotFound) {
                 [self->filteredColleges addObject:college];
             } else if ([college.details rangeOfString:searchText].location != NSNotFound) {
                 [self->filteredColleges addObject:college];
             }
         }
-        NSString *correctWord = [AutocorrectFunctions findCorrectWord:searchText forCollegesInArray:self->colleges];
+        NSString *correctWord = [AutocorrectFunctions findCorrectWord:searchText forCollegesInArray:self->collegesAtSegment];
         self->correctWordToDisplayInSearchBar = correctWord;
     }
     [self.tableView reloadData];
 }
 
+- (IBAction)didTapRigor:(id)sender {
+    NSSortDescriptor *sortingBasedOnRigor = [[NSSortDescriptor alloc] initWithKey:@"rigorScore" ascending:YES];
+    self->collegesAtSegment = [self->collegesAtSegment sortedArrayUsingDescriptors:@[sortingBasedOnRigor]];
+    [self.tableView reloadData];
+}
+
+
 - (void)searchBarSearchButtonClicked:(UISearchBar *)searchBar {
     searchBar.text = self->correctWordToDisplayInSearchBar;
 }
 
+- (void)getCollegesForSegmentControl {
+    NSString *segmentPosition = [NSString stringWithFormat:@"%ld", (self.segmentControlHome.selectedSegmentIndex*10 +10)];
```

Comment:
lets add a descriptive variable for self.segmentControlHome.selectedSegmentIndex*10 +10 !